src/Snapshot.php

```diff
@@ -61,6 +61,103 @@ public function load(string $connectionName = null)
         event(new LoadedSnapshot($this));
     }
 
+    public function loadStream(string $connectionName = null)
+    {
+        event(new LoadingSnapshot($this));
+
+        if ($connectionName !== null) {
+            DB::setDefaultConnection($connectionName);
+        }
+
+        $this->dropAllCurrentTables();
+
+        if ($this->compressionExtension === 'gz') {
+            $dumpFilePath = $this->decompress();
+        } else {
+            $dumpFilePath = $this->disk->path($this->fileName);
+        }
+
+        $this->streamFileIntoDB($dumpFilePath, $connectionName);
+
+        event(new LoadedSnapshot($this));
+    }
+
+    public function streamFileIntoDB($path, string $connectionName = null)
+    {
+        if ($connectionName !== null) {
+            DB::setDefaultConnection($connectionName);
+        }
+
+        $tmpLine = '';
+        $lines = file($path);
+        $errors = [];
+
+        foreach ($lines as $line) {
```

Comment:
Can we use a collection method here?

Reply:
How would I do this? Should I run `collect()` the `$lines` and then use the collection methods? I'm not sure that works in regard to memory. I can give it a try.

Reply:
I suggest that we refactor this to a Lazy Collection method which is supported in Laravel 6+. This is taken from the [Lazy Collections documentation](https://laravel.com/docs/7.x/collections#creating-lazy-collections)

```php
use Illuminate\Support\LazyCollection;

LazyCollection::make(function () {
    $handle = fopen('log.txt', 'r');

    while (($line = fgets($handle)) !== false) {
        yield $line;
    }
});
```

probably a better way is streaming the content line for line exist using the FlySystem:
```php
$stream = $this->disk->readStream($path);
```
we can then refactor this to a function for streaming the content SQL line for line, meaning line ending with a semicolon.